lib/rex/proto/kerberos/pac/krb5_pac.rb

```diff
@@ -589,20 +603,42 @@ def assign(val)
     end
 
     # Calculates the checksums, can only be done after all other fields are set
-    def calculate_checksums!(key: nil)
+    # @param [String] key Service key to calculate the server checksum
+    # @param [String] krbtgt_key key to calculate priv server (KDC) and full checksums with, if not specified `key` is used
+    def calculate_checksums!(key: nil, krbtgt_key: nil)
```

Comment:
```suggestion
    # @param [String] service_key Service key to calculate the server checksum
    # @param [String] krbtgt_key key to calculate priv server (KDC) and full checksums with, if not specified `key` is used
    def calculate_checksums!(service_key: nil, krbtgt_key: nil)
```


Let's also add links to the spec here for the future folk:

https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-kile/962edb93-fa3c-48ea-a0a6-062f760eb69c

https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-kile/37bf87b9-1d56-475a-b2b2-e3948b29194d

Reply:
Pre-emptive: Since there isn't official docs so far for the new field, can we link to the samba issue + BlackHat PDF